tests/test_objective_funs.py

```diff
@@ -570,6 +577,178 @@ def test(eq):
         test(get("HELIOTRON"))
 
 
+class TestQuadraticFlux:
+    """Tests quadratic flux."""
+
+    def get_grids(self, eq, field, N_grid=15, M_grid=10):
+        """Compute grids used for all of the tests."""
+        field_grid = LinearGrid(
+            rho=np.array([1.0]),
+            M=M_grid,
+            N=N_grid,
+            NFP=field.NFP if not isinstance(field, ToroidalMagneticField) else 1,
+            sym=False,
+        )
+        eval_grid = LinearGrid(
+            rho=np.array([1.0]),
+            M=eq.M_grid,
+            N=N_grid,
+            NFP=eq.NFP,
+            sym=False,
+        )
+
+        source_grid = LinearGrid(
+            rho=np.array([1.0]),
+            M=M_grid,
+            N=N_grid,
+            NFP=int(eq.NFP),
+            sym=False,
+        )
+
+        return field_grid, eval_grid, source_grid
+
+    def make_nonaxisym_field(self, eq):
+        """Makes a nonaxisymmetric field depending on the given eq."""
+        surf = eq.surface.copy()
+        surf.R_lmn = surf.R_lmn.at[surf.R_basis.get_idx(M=1, N=0)].set(2)
+        surf.Z_lmn = surf.Z_lmn.at[surf.Z_basis.get_idx(M=-1, N=0)].set(-2)
+        nonaxisym_field = FourierCurrentPotentialField.from_surface(
+            surface=surf,
+            Phi_mn=np.array([1e4]),
+            modes_Phi=np.array([[1, 0]]),
+            G=1e4,
+            I=0,
+            sym_Phi="sin",
+        )
+
+        return nonaxisym_field
+
+    @pytest.mark.unit
+    def test_quadratic_flux(self, quadratic_flux_equilibriums):
+        """Test calculation of quadratic flux on the boundary."""
+        t_field = ToroidalMagneticField(1, 1)
+
+        # test that torus (axisymmetric) Bnorm is exactly 0
+        __, eq = quadratic_flux_equilibriums
+        obj = QuadraticFlux(eq, t_field)
+        obj.build(eq, verbose=2)
+        f = obj.compute(field_params=t_field.params_dict)
+        np.testing.assert_allclose(f, 0, rtol=1e-15, atol=1e-15)
+
+        # test nonaxisymmetric surface
+        eq = desc.examples.get("precise_QA", "all")[0]
+        eq.change_resolution(4, 4, 4, 8, 8, 8)
+
+        __, eval_grid, source_grid = self.get_grids(eq, t_field)
+
+        obj = QuadraticFlux(
+            eq,
+            t_field,
+            eval_grid=eval_grid,
+            src_grid=source_grid,
+        )
+        Bnorm = t_field.compute_Bnormal(
+            eq.surface, eval_grid=eval_grid, source_grid=source_grid
+        )[0]
+        obj.build(eq)
+        f = obj.compute(field_params=t_field.params_dict)
+
+        np.testing.assert_allclose(f, Bnorm, atol=1e-3)
+
+    @pytest.mark.unit
+    def test_quadratic_flux_with_eq_fixed(self, quadratic_flux_equilibriums):
+        """Test with eq_fixed = True, field_fixed = False.
+
+        Checks that a non-axisymmetric field becomes axisymmetric
+        given an axisymmetric field.
```

Comment:
This says is uses `eq_fixed=True` but it doesn't actually. 

Also the second sentence of the description doesn't make sense.